### PR TITLE
Remove deprecated EventStore based I/O functionality

### PR DESCRIPTION
**Pull request by tmadlener**
BEGINRELEASENOTES
- **Remove the deprecated EventStore functionality as announced in #429**

ENDRELEASENOTES

~This is currently only the first steps to remove the EventStore based I/O functionality (as outlined in #429). It is quite far from complete. The main purpose at this point is to get a feeling of how many things need to be touched in the end.~ This removes all traces of the `EventStore` and related readers and writers.

- [x] Needs #486 
- [x] Pre-release (v00-17-01)
- [x] Remove python bindings 
- [x] Fix tests
  - [x] Make unittests independent of EventStore
  - [x] Make some of the I/O tests use Frame based I/O
  - [x] Cleanup python tests
- [x] Cleanup documentation
- [x] #505 
- [x] Needs https://github.com/key4hep/k4FWCore/pull/171

### CODE REVIEW

**Comment by tmadlener** [comment]
What is left to do here is essentially checking the documentation for remnants of the EventStore but we could also address that in fixing the documentation for version 1.0.

File: doc/examples.md [hunk 1]
```diff
@@ -130,51 +132,20 @@ Or implicitly when following an object reference. In both cases the access to da
...
+example of reading files looks like this:
+
 
 ```python
-    from EventStore import EventStore
-    store = EventStore(<list of files>)
-    for event in store:
+    from podio.root_io import Reader
+    
+    reader = Reader("one or many input files")
```
Comment (tmadlener) [inline doc/examples.md:147]:
```suggestion
    reader = Reader("one or many input files")
```

**Comment by tmadlener** [comment]
Now, I think I got everything EventStore related including documentation.

File: doc/examples.md [hunk 2]
```diff
@@ -130,51 +132,20 @@ Or implicitly when following an object reference. In both cases the access to da
...
+provide all the necessary functionality to read and write event files. An
+example of reading files looks like this:
+
 
 ```python
-    from EventStore import EventStore
-    store = EventStore(<list of files>)
-    for event in store:
+    from podio.root_io import Reader
+
```
Comment (tmadlener) [inline doc/examples.md:146]:
```suggestion

```

File: doc/examples.md [hunk 3]
```diff
@@ -130,51 +132,19 @@ Or implicitly when following an object reference. In both cases the access to da
 Sometimes it is necessary or useful to store additional data that is not directly foreseen in the EDM.
 This could be configuration parameters of simulation jobs, or parameter descriptions like cell-ID encoding etc. PODIO currently allows to store such meta data in terms of a `GenericParameters` class that
 holds an arbitrary number of named parameters of type `int, float, string` or vectors if these.
-Meta data can be stored and retrieved from the `EventStore` for runs, collections and events via
-the three methods:
-```cpp
-virtual GenericParameters& EventStore::getRunMetaData(int runID);
```
Comment (hegner) [inline doc/examples.md:136]: should we add a small explanation that one just creates a frame if there is a metadata use case? Just to explain a bit more the frame concept

Comment (hegner) [reply doc/examples.md:136]: for a non-default one where putting a parameter is not enough

**Review by hegner — APPROVED** [review]
I think we have to quickly come back to the metadata interface discussion like #407 and #228 before cutting the release. Otherwise fine.
Should we wait for Monday/Tuesday for the merge? We may kill nightlies over weekend otherwise

File: doc/examples.md (hunk 3 above)
Comment (hegner) [reply doc/examples.md:136]: That can go to a separate PR

**Comment by tmadlener** [comment]
Not entirely sure what to make of CI at the moment. Things that fail in CI are working locally, and things that fail locally are failing in CI...

File: doc/advanced_topics.md [hunk 4]
```diff
@@ -32,26 +32,14 @@ Before writing out a collection, the data need to be put into the proper structu
...
-    collection->setID( <collection ID read from disk> );
-    collection->prepareAfterRead();
-    // ...
-    collection->setReferences( &collectionProvider );
-```
-
-The strong assumption here is that all references are being followed up directly and no later on-demand reading is done.
+The main requirement for a reading backend is its capability of reading back all
+the necessary data from which a collection can be constructed in the form of
+`podio::CollectionReadBuffers`. From thes buffers collections can then be
```
Comment (tmadlener) [inline doc/advanced_topics.md:37]:
```suggestion
`podio::CollectionReadBuffers`. From these buffers collections can then be
```

File: tests/CMakeLists.txt [hunk 5]
```diff
@@ -94,4 +100,4 @@ endif()
 # Customize CTest to potentially disable some of the tests with known problems
 configure_file(CTestCustom.cmake ${PROJECT_BINARY_DIR}/CTestCustom.cmake @ONLY)
 
-ExternalData_Add_Target(legacy_test_cases)
+message(STATUS "Test inputs will be stored in: ${ExternalData_OBJECT_STORES} if they are not already present")
```
Comment (tmadlener) [inline tests/CMakeLists.txt:103]: This should go to the top-level `CMakeLists.txt` probably.

**Pull request merged**